### DISCUSSION TITLE
Resize window height (without changing its width)

### DISCUSSION THREAD

**Discussion by torgeir** [rxhanson/Rectangle#387] [Ideas]
I would love the ability to make windows "just a tad longer". A couple of shortcuts similar to Smaller and Larger but that only works in the vertical direction would be a great addition to this already awesome application! 🎉

**Comment by jli** [comment]
+1 I'd love shortcuts for increase/decrease window height (and perhaps also increase/decrease window width) which behaves just like the existing increase/decrease size, just in 1 dimension.

@rxhanson I think this wouldn't be too hard to add and I'd be keen to give it a shot if you're supportive? I guess the main issue is that it would add more items into the UI, though there is already an overflow section so it wouldn't add much clutter to the default view.

**Reply by rxhanson** [reply]
@jli, go for it. In terms of the UI, I think I'm finally ready to just port over the list view from Hookshot. For the longest time, I've felt like Rectangle should reflect the simple UI of Spectacle, but I personally prefer the list view and I'm tired of fighting the addition of more actions just to keep the simple UI. I'll be looking at switching over the UI in the coming weeks - you can either wait until that's done or go ahead and add it and we'll handle the merge later.

**Reply by jli** [reply]
@rxhanson Thanks. I overcame some M1 issues and got XCode building successfully.

Could you give me some general pointers about what code locations are relevant? 
- where to add another entry to the keybindings UI
- where to add the additional window sizing behavior

**Reply by rxhanson** [reply]
>  where to add another entry to the keybindings UI

Main.storyboard, PrefsViewController, WindowAction

>  where to add the additional window sizing behavior

WindowAction, WindowCalculation (add a new Calculation class)

**Reply by mihaiconstantin** [reply]
I think this would be a killer addition to Rectangle. The possibility of independently adjusting the height and width gives you room to create pretty much whatever configuration you want. Is there any progress on this already?

> I've felt like Rectangle should reflect the simple UI of Spectacle, but I personally prefer the list view and I'm tired of fighting the addition of more actions just to keep the simple UI.

@rxhanson I find the functionality more important than the UI, especially since most of the functionality of Rectangle is keystroke-based and only a limited amount of time is spent in the UI. Hence, to me, a list view sounds pretty good. 

This might not be entirely related, but have you seen how they handle the UI for creating custom layouts in PowerToys (e.g., https://docs.microsoft.com/en-us/windows/powertoys/fancyzones#choose-your-layout-layout-editor)?

**Reply by Skoti** [reply]
The current increase/decrease size behavior depends on wether the window is touching screen edges or not.
So if it's full height, it will only increase/decrease its width. If it's full width it will only increase/decrease its height.
And if it's neither full height nor full width it will increase/decrease its size in both dimensions (ofc until it becomes full width or height).

I would love to see the option to configure the shortcut to l increase/decrease size in a horizontal/vertical/diagonal direction independently of whether the window is full width/height.

**Reply by bztravis** [reply]
+1, is this feature still being worked on?

**Reply by rxhanson** [reply]
Been a while since I've thought about this. I never did end up adding the list view from Rectangle Pro into Rectangle, in the interest of keeping it simpler (in hindsight, a good decision). However, if anyone wants to add these in as hidden, terminal command enabled, shortcuts I see no reason to turn down a good pull request.
